### ISSUE TITLE
Support VTT in Matroska

### ISSUE THREAD

**Issue by moneytoo** [google/ExoPlayer#9886]
I'm looking into adding support for WebVTT embedded in Matroska (MKV). 

I managed to get basic support in this patch: https://github.com/moneytoo/ExoPlayer/commit/09b07da83fb9cf6a0a6e511e7c0f6282c963e1da and it seems to be working with the sample provided to me in here: https://github.com/moneytoo/Player/issues/249

The patch still lacks support for styles as described in [Matroska specs](https://www.matroska.org/technical/subtitles.html#webvtt). I'm planning on looking into the styles as well, but I'm pretty busy lately so I'm wondering if I should send you PR with this basic support (is it worth an attempt?) or better wait and finish it all?

**Comment by icbaker** [comment]
I think sending a PR with what you've got so far makes sense.

**Heads up:** We have some work planned that will change how subtitle data flows from muxed formats like MKV to the `TextRenderer` - and that will likely affect this issue too.

**Some background:** Currently we put samples in the `SampleQueue` containing data in the 'original' subtitle format, and then parse/decode it inside the `TextRenderer`. We recently added support for parsing the subtitle data **before** we put samples into the `SampleQueue`, which brings various advantages and simplifications. This currently only works for sideloaded subtitle files (those specified in `MediaItem.localConfiguration.subtitleConfigurations`), and is opt-in (controlled by [`DefaultMediaSourceFactory#experimentalUseProgressiveMediaSourceForSubtitles`](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/source/DefaultMediaSourceFactory.html#experimentalUseProgressiveMediaSourceForSubtitles(boolean))). We are planning to standardise all subtitle-processing to follow this model, which (among other changes) will involve changing `MatroskaExtractor` to parse the subtitle data it reads from each sample as it goes along.

If you get the PR in before we start that work in `MatroskaExtractor` then it will help to ensure we don't somehow make this improvement completely impossible in the new architecture :)

**Comment by icbaker** [comment]
I'm going to mark this as low priority to indicate that we (exoplayer-team) have no immediate plans to directly work on implementing this - but we will of course prioritise reviewing and merging any PRs you send us!

**Comment by marek22k** [comment]
I would also like this feature. yt-dlp, for example, downloads all subtitles in vtt format. I have the following options for watching the videos with subtitles:
1. convert subtitles to .srt with ffmpeg.
2. use VLC Media Player.

Both options are not really nice.
